dry_rest_permissions/generics.py

```diff
@@ -120,14 +124,35 @@ def has_permission(self, request, view):
             if hasattr(model_class, action_method_name):
                 return getattr(model_class, action_method_name)(request)
 
-        if request.method in permissions.SAFE_METHODS:
-            assert hasattr(model_class, 'has_read_permission'), \
-                self._get_error_message(model_class, 'has_read_permission', action_method_name)
-            return model_class.has_read_permission(request)
-        else:
-            assert hasattr(model_class, 'has_write_permission'), \
-                self._get_error_message(model_class, 'has_write_permission', action_method_name)
-            return model_class.has_write_permission(request)
+        try:
+            if request.method in permissions.SAFE_METHODS:
+                assert hasattr(model_class, 'has_read_permission'), \
+                    self._get_error_message(model_class, 'has_read_permission', action_method_name)
+                return model_class.has_read_permission(request)
+            else:
+                assert hasattr(model_class, 'has_write_permission'), \
+                    self._get_error_message(model_class, 'has_write_permission', action_method_name)
+                return model_class.has_write_permission(request)
+        except AssertionError as e:
+            if self.django_permissions:
+                django_permission, django_permission_exists = None, False
+                if hasattr(view, 'action'):
+                    action = self._get_action(view.action)
+                    django_actions_map = {'list': 'view', 'retrieve': 'view', 'update': 'change',
+                                          'destroy': 'delete', 'create': 'add'}
```

Comment:
I'm not sure here but maybe check the documentation to be sure we don't need to include `partial_update` too

Reply:
As far as I know, Django only has [4 default permissions](https://docs.djangoproject.com/en/3.1/topics/auth/default/#default-permissions).
It doesn't make sense to add duplicate entries since it will just fall back to `django_generic_action` which would use `view` or `change` depending on `SAFE_METHODS` which should be set correctly for `partial_update`.

I'm kinda tempted to remove `list` as well to allow users to define custom `list_model` permissions separate from the `view`/`retrive`.

Reply:
Hum, i think you hit a limitation of the PR's idea here:

1. Django permission does not offer a clear way to define which permission is for what. It's just a list of `key` we can use where we want and how we want.

2. People can create other `key` manually or with Django-Guardian and there is no standard of naming.

3. Peope can create a lot of different structure of permissions check since no structure is gave by default with Django or Django-Guardian.

---

If people use Django-Guardian maybe they created new custom set of key as you described (ie: `list_model`) but even in this case we will not have a standard architecture to build our implementation on since the key can have the name of their choice.

Trying to check the permissions based on the keys are maybe not the good way to go, maybe the good way to do it would be to give users an easy way to check the DryRestPermissions to be able to integrate it in their actual custom permissions architecture.

What do you think ?

Reply:
An other approach could be to allow users define a structure of permission they gave in argument or in settings, to define which permission `key` is for what. But it will match only a subset of the possible architecture actually possible with Django Permission since it's completelly architecture free

Reply:
I was personally going for convention over configuration approach. Since it is very easy to overwrite the "guessed" permission names by creating an explicit `has_x_permission` method calling `request.user.has_perm` with any name as stated in the opening comment, I don't see the point of adding extra configuration in the view (which would be repeated) vs adding a method in the model.

As for standard, I was thinking `{action}_{model}`. So for example, if you had a model `Category` with a custom action `auto_complete` on your DRF ModelViewSet, my code would check for the existence of permission named `app.auto_complete_category`, or `app.view_category` (assuming the custom action was using `GET` method).

Reply:
> If people use Django-Guardian maybe they created new custom set of key as you described

There is no need for extra libraries, Django itself allows adding new permissions very easily through the `Meta` section of models: https://docs.djangoproject.com/en/3.1/ref/models/options/#permissions

While there is no explicit structure for it, you can write a description. For my previous example, the user would add this line to the `Category` model's meta section:
```python
permissions = [('auto_complete_category', 'Can view list of categories for auto-completing select tags')]
```
knowing that the name is DRF's action name + model's name, I think it is intuitive if the docs explain it well.

Reply:
I understand your point, i'm just thinking that in a real case when somebody will add DryRestPermissions on top of it's actual architecture, he will have only little chances to have exactly your convention and so the integration will not work.

Also i'm a bit scared that the developer will maybe think that everything is fine while in reality the check we do is really unofficial and have big chance to not match his own code convention (being optimistic that he had one that he respected all the time and tested). 

That said i think that your solution help to fix a concrete need and it's better to start with your implementation than doing nothing, and we will have the time to see how developer use it and how we can enhance it. Just make sure that documentation is really clear on how the feature work and what need to be take in consideration to keep security on existing project.

Go for it :+1: